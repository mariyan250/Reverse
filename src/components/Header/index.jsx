// React and Style
import React from 'react';
import styles from './index.module.scss';

// Redux
import { connect } from 'react-redux';
import { toggleHamburger, closeHamburger } from 'store/actions/hamburger';
import { removeUser } from 'store/actions/user';

// Translation
import { useTranslation } from 'react-i18next';

// Router and Routes
import { NavLink, Link, useHistory } from 'react-router-dom';
import { routes } from 'constants/routes';

// Components
import Hamburger from 'components/Hamburger';
import Menu from 'components/Menu';

function Header(props) {
  const { t } = useTranslation();
  const history = useHistory();

  const publicLinks = [
    { to: routes.home, text: 'header.links.home' },
    { to: routes.login, text: 'header.links.login' },
    { to: routes.register, text: 'header.links.register' },
  ];

  const privateLinks = [
    { to: routes.home, text: 'header.links.home' },
    { to: routes.profile, text: 'header.links.profile' },
  ];

  const handleLogout = () => {
    props.removeUser();
    history.push('/');
  };

  return (
    <header className={`${styles.header} app-bg-secondary position-sticky`}>
      <div
        className={`${styles['header-content']} d-flex d-flex justify-content-between align-items-center h-100 mx-auto px-4 px-md-5`}
      >
        <Link to="/">
          <span className="app-text-white">{t('header.heading')}</span>
        </Link>

        <ul className="d-none d-md-flex justify-content-between align-items-center h-100 m-0">
          {(props.user ? privateLinks : publicLinks).map((link, i) => (
            <li className="ml-5" key={i}>
              <NavLink
                to={link.to}
                className="d-flex justify-content-center align-items-center app-text-white"
                activeClassName={styles.active}
              >
                {t(link.text)}
              </NavLink>
            </li>
          ))}

          {props.user && (
            <li className="ml-5">
              <button
                onClick={handleLogout}
                className="border-0 bg-transparent app-text-white d-flex align-items-end"
                data-testid="logout"
                aria-label="logout button"
              >
                <i className="fas fa-sign-out-alt h-100" />
              </button>
            </li>
          )}
        </ul>

        <div className="d-md-none">
          <Hamburger
            onClick={() => props.toggleHamburger(!props.burgerOpened)}
            isOpen={props.burgerOpened}
          />
        </div>

        {props.burgerOpened && (
          <div className={`${styles.menu} d-md-none position-fixed w-100`}>
            <Menu links={props.user ? privateLinks : publicLinks} />
          </div>
        )}
      </div>
    </header>
  );
}

const mapStateToProps = (state) => ({
  burgerOpened: state.hamburger.isOpen,
  user: state.user,
});

export default connect(mapStateToProps, { toggleHamburger, removeUser, closeHamburger })(Header);
