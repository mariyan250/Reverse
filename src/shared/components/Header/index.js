// React and Style
import React from 'react';
import './index.scss';

// Router and Routes
import { NavLink, Link } from 'react-router-dom';
import { routes } from 'shared/constants/routes';

// Redux
import { connect } from 'react-redux';
import { toggleHamburger } from 'store/actions/hamburger';

// Translation
import { useTranslation } from 'react-i18next';

// Components
import Hamburger from 'shared/components/Hamburger';
import Menu from 'shared/components/Menu';

function Header(props) {
  const { t } = useTranslation();

  const links = [
    { to: routes.dashboard, text: 'header.links.home' },
    { to: routes.login, text: 'header.links.login' },
    { to: routes.register, text: 'header.links.register' },
  ];

  return (
    <header className="app-bg-secondary position-sticky px-4 px-md-5 d-flex justify-content-between align-items-center">
      <Link to="/">
        <h1 className="app-text-white">{t('header.heading')}</h1>
      </Link>

      <ul className="d-none d-md-flex h-100 align-items-center m-0">
        {links.map((link) => (
          <li className="ml-5 h-100">
            <NavLink
              to={link.to}
              className="app-text-white h-100 d-flex justify-content-center align-items-center"
            >
              {t(link.text)}
            </NavLink>
          </li>
        ))}
      </ul>

      <Hamburger
        className={`d-md-none ${props.burgerOpened ? 'toggled' : ''}`}
        onClick={() => props.toggleHamburger(!props.burgerOpened)}
      />

      {props.burgerOpened && (
        <Menu className="header-menu position-fixed d-md-none w-100" links={links} />
      )}
    </header>
  );
}

const mapStateToProps = (state) => {
  return {
    burgerOpened: state.hamburger.isOpen,
  };
};

export default connect(mapStateToProps, { toggleHamburger })(Header);
