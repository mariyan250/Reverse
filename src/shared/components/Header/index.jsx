// React and Style
import React from 'react';
import styles from './index.module.scss';

// Redux
import { connect } from 'react-redux';
import { toggleHamburger, closeHamburger } from 'store/actions/hamburger';
import { removeUser } from 'store/actions/user';

// Translation
import { useTranslation } from 'react-i18next';

// Router and Routes
import { NavLink, Link } from 'react-router-dom';
import { routes } from 'shared/constants/routes';

// Components
import Hamburger from 'shared/components/Hamburger';
import Menu from 'shared/components/Menu';

function Header(props) {
  const { t } = useTranslation();

  const publicLinks = [
    { to: routes.home, text: 'header.links.home' },
    { to: routes.login, text: 'header.links.login' },
    { to: routes.register, text: 'header.links.register' },
  ];

  const privateLinks = [
    { to: routes.home, text: 'header.links.home' },
    { to: routes.profile, text: 'header.links.profile' },
  ];

  const handleLogout = () => {
    props.removeUser();
    window.location = '/';
  };

  return (
    <header
      className={`${styles.header} d-flex justify-content-between align-items-center app-bg-secondary position-sticky px-4 px-md-5`}
    >
      <Link to="/">
        <span className="app-text-white">{t('header.heading')}</span>
      </Link>

      <ul className="d-none d-md-flex h-100 align-items-center m-0">
        {(props.user ? privateLinks : publicLinks).map((link, i) => (
          <li className="ml-5 h-100" key={i}>
            <NavLink
              to={link.to}
              className="d-flex justify-content-center align-items-center app-text-white h-100"
            >
              {t(link.text)}
            </NavLink>
          </li>
        ))}

        {props.user && (
          <li
            className="ml-3 py-2 px-3 app-bg-secondary app-text-white border-0"
            onClick={handleLogout}
            data-testid="logout"
          >
            {t('header.logout_btn')}
          </li>
        )}
      </ul>

      <div className="d-md-none">
        <Hamburger
          onClick={() => props.toggleHamburger(!props.burgerOpened)}
          isOpen={props.burgerOpened}
        />
      </div>

      {props.burgerOpened && (
        <Menu
          className={`${styles.menu} position-fixed d-md-none w-100`}
          links={props.user ? privateLinks : publicLinks}
        />
      )}
    </header>
  );
}

const mapStateToProps = (state) => {
  return {
    burgerOpened: state.hamburger.isOpen,
    user: state.user,
  };
};

export default connect(mapStateToProps, { toggleHamburger, removeUser, closeHamburger })(Header);
