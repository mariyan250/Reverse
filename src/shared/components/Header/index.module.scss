@import '../../../index.scss';

.header {
  height: to-rem(60px);
  z-index: 10;
  top: 0;

  & > a {
    span {
      font-size: to-rem(32px);
    }
  }

  ul {
    li {
      font-size: to-rem(16px);

      button {
        i {
          font-size: to-rem(24px);
        }
      }
    }
  }

  .menu {
    top: to-rem(60px);
    left: 0;
    z-index: 10;
    height: 10rem;
    animation: appear 1 $timing-slow;

    button {
      outline: none;
    }
  }
}
