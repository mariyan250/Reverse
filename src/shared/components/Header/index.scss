@import '../../../index.scss';

header {
  height: to-rem(60px);
  z-index: 10;
  top: 0;

  ul {
    a {
      font-size: to-rem(16px);
    }
  }

  .header-menu {
    top: to-rem(60px);
    left: 0;
    z-index: 10;
    animation: appear 1 $slow;
  }
}
