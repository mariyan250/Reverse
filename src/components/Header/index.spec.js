import React from 'react';
import { render } from 'tests/test-utils';
import Header from './index';

describe('Header', () => {
  it('should render the component', () => {
    const { container } = render(<Header />);
    expect(container).toMatchSnapshot();
  });
});
