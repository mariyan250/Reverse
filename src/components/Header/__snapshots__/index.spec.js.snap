// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render the component 1`] = `
<div>
  <header
    class="header d-flex justify-content-between align-items-center app-bg-secondary position-sticky px-4 px-md-5"
  >
    <a
      href="/"
    >
      <span
        class="app-text-white"
      >
        header.heading
      </span>
    </a>
    <ul
      class="d-none d-md-flex justify-content-between align-items-center h-100 m-0"
    >
      <li
        class="ml-5"
      >
        <a
          aria-current="page"
          class="d-flex justify-content-center align-items-center app-text-white active"
          href="/"
        >
          header.links.home
        </a>
      </li>
      <li
        class="ml-5"
      >
        <a
          class="d-flex justify-content-center align-items-center app-text-white"
          href="/login"
        >
          header.links.login
        </a>
      </li>
      <li
        class="ml-5"
      >
        <a
          class="d-flex justify-content-center align-items-center app-text-white"
          href="/register"
        >
          header.links.register
        </a>
      </li>
    </ul>
    <div
      class="d-md-none"
    >
      <button
        aria-label="toggle menu"
        class="hamburger false d-flex flex-column justify-content-between border-0 bg-transparent"
        data-testid="hamburger"
      >
        <span
          class="app-bg-white rounded-pill"
        />
        <span
          class="app-bg-white align-self-end rounded-pill"
        />
        <span
          class="app-bg-white align-self-end rounded-pill"
        />
      </button>
    </div>
  </header>
</div>
`;
